app/code/core/Mage/Adminhtml/Helper/TwoFactorAuthentication.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/**
+ * Maho
+ *
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ * @copyright  Copyright (c) 2024 Maho (https://mahocommerce.com)
+ * @license    https://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+/**
+ * @category   Mage
+ * @package    Mage_Adminhtml
+ */
+class Mage_Adminhtml_Helper_TwoFactorAuthentication extends Mage_Core_Helper_Abstract
+{
+    public function getSecret(): string
+    {
+        return \OTPHP\TOTP::create()->getSecret();
+    }
+
+    public function getQRCode(#[\SensitiveParameter] string $username, #[\SensitiveParameter] string $secret): string
+    {
+        $storeName = Mage::getStoreConfig('general/store_information/name');
+        $otp = \OTPHP\TOTP::create($secret);
+        $otp->setLabel($username . '@' . $storeName);
+        $otp->setIssuer('Maho Admin');
```

Comment:
I didn't have a store name set, so it showed as `Maho Admin: admin@`

```suggestion
        $otp = \OTPHP\TOTP::create($secret);
        $otp->setIssuer('Maho Admin');
        if ($storeName = Mage::getStoreConfig('general/store_information/name')) {
            $otp->setLabel($username . '@' . $storeName);
        } else {
            $otp->setLabel($username);
        }
```

Reply:
Maybe also an image field? It's non standard, but some apps support it:

```php
$otp->setParameter('image', 'https://placehold.co/400x400.png');
```

Note I couldn't even find a publicly hosted Maho logo in PNG format since everything is SVG. 😅

Reply:
@justinbeaty about the image, does it work for you? cause on Authy it's not working so I don't know, I wouldn't add it also because maybe people would like to have their own store logo?

Reply:
The image does work on FreeOTP which I downloaded to test out. On Authy I get some weird image, I assume you get the same. Apparently they use a result from Google Image Search. It seems they have a way to ask their support team to use the proper logo, but that has nothing to do with code.

I think even on a customer's store, Authy is still going to google search "Maho Admin" since that's set as the issuer. Maybe if we made it:

```php
if ($storeName = Mage::getStoreConfig('general/store_information/name')) {
    $otp->setIssuer($storeName);
} else {
    $otp->setIssuer('Maho Admin');
}

$otp->setLabel($username);
```

Reply:
done! :-)